include/GoogleSync/GoogleSyncBase.php

```diff
@@ -0,0 +1,1049 @@
+            $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'The Google Client is not set up. See setClient Method');
+            return false;
+        }
+
+        // create new calendar service
+        $this->gService = new Google_Service_Calendar($this->gClient);
+        if ($this->isServiceExists()) {
+            return true;
+        }
+        // TODO: Confusing return value: It should be an exception, do not use similar return value (false) by different reaon os failure (see above)
+        $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Setting $this->gService Failed');
+        return false;
+    }
+
+    /**
+     * Push event from SuiteCRM to Google Calendar
+     *
+     * If the google event is not provided, a new one will be created
+     * and inserted. If one is provided, the existing Google Event will
+     * be updated.
+     *
+     * @param Meeting $event_local : SuiteCRM Meeting Bean
+     * @param \Google_Service_Calendar_Event $event_remote (optional) \Google_Service_Calendar_Event Object
+     *
+     * @return string|bool Meeting Id on success, false on failure
+     */
+    protected function pushEvent(Meeting $event_local = null, Google_Service_Calendar_Event $event_remote = null)
+    {
+        if (!$event_local instanceof Meeting) {
+            throw new InvalidArgumentException('Argument 1 passed to GoogleSyncBase::pushEvent() must be an instance of Meeting, ' . getType($event_local) . ' given.');
+        }
+        if (!$this->gService instanceof Google_Service_Calendar) {
+            throw new GoogleSyncException('GooleSyncBase is trying to push event but Google_Service_Calendar_Resource_Events is not set.', GoogleSyncException::NO_GSERVICE_SET);
+        }
+        if (!$this->gService->events instanceof Google_Service_Calendar_Resource_Events) {
+            throw new GoogleSyncException('GooleSyncBase is trying to push event but Google_Service_Calendar_Resource_Events is not set.', GoogleSyncException::NO_GRESOURCE_SET);
+        }
+        
+        if (!isset($event_remote) || empty($event_remote)) {
+            $event = $this->createGoogleCalendarEvent($event_local);
+            $return = $this->gService->events->insert($this->calendarId, $event);
+        } elseif (isset($event_remote)) {
+            $event = $this->updateGoogleCalendarEvent($event_local, $event_remote);
+            $return = $this->gService->events->update($this->calendarId, $event->getId(), $event);
+        }
+
+        /* We don't get a status code back showing success. Instead, the return of the
+         * create or update is the Google_Service_Calendar_Event object after saving.
+         * So we check to make sure it has an ID to determine Success/Failure.
+         */
+        if (!isset($return->id)) {
+            // TODO: Confusing return value: It should be an exception, do not use similar return value (false) by different reaon os failure (see belove)
```

Comment:
@gymad 
I think I could throw and then catch it in doAction(). Then I could replace the exception message there with the message/code from this.

Reply:
Fixed